go/libraries/doltcore/sqle/dsess/dolt_session.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dsess
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+
+	"github.com/dolthub/go-mysql-server/sql"
+
+	"github.com/dolthub/dolt/go/cmd/dolt/cli"
+	"github.com/dolthub/dolt/go/libraries/doltcore/env"
+	"github.com/dolthub/dolt/go/libraries/utils/config"
+)
+
+type DoltSession struct {
+	*Session
+	globalsConf config.ReadWriteConfig
+	mu          *sync.Mutex
+}
+
+var _ sql.Session = (*DoltSession)(nil)
+var _ sql.PersistableSession = (*DoltSession)(nil)
+
+func NewDoltSessionFromDefault(sess *Session, conf config.ReadWriteConfig) *DoltSession {
```

Comment:
FromDefault doesn't really make sense here

WithConfig?

Reply:
this inputs a`NewDefaultSession`, but i see how that can be confusing. How about `dsess.Session.NewDoltSession(conf)`?